lightly/utils/cropping/crop_image_by_bounding_boxes.py

```diff
@@ -71,19 +59,28 @@ def crop_dataset_by_bounding_boxes_and_save(dataset: LightlyDataset,
         filepath_image = dataset.get_filepath_from_filename(filename_image)
         filepath_image_base, image_extension = os.path.splitext(filepath_image)
 
-        filepath_out_dir = os.path.join(output_dir, filename_image).replace(image_extension, '')
+        filepath_out_dir = os.path.join(output_dir, filename_image)\
+            .replace(image_extension, '')
         Path(filepath_out_dir).mkdir(parents=True, exist_ok=True)
 
-        cropped_images = crop_image_by_bounding_boxes(filepath_image, bounding_boxes)
+        image = Image.open(filepath_image)
+        
         cropped_images_filepaths = []
-        for index, (class_index, cropped_image) in enumerate((zip(class_indices, cropped_images))):
+        for index, (class_index, bbox) in \
+                enumerate((zip(class_indices, bounding_boxes))):
             if class_names:
                 class_name = class_names[class_index]
             else:
                 class_name = f"class{class_index}"
             cropped_image_last_filename = f'{index}_{class_name}{image_extension}'
             cropped_image_filepath = os.path.join(filepath_out_dir, cropped_image_last_filename)
+
+            w, h = image.size
+            crop_box = (w * bbox.x0, h * bbox.y0, w * bbox.x1, h * bbox.y1)
+            crop_box = tuple(int(i) for i in crop_box)
+            cropped_image = image.crop(crop_box)
             cropped_image.save(cropped_image_filepath)
+            del cropped_image
```

Comment:
I think this is not necessary here as we reuse the same variable in the next loop iteration anyway